src/Simplify.cpp

```diff
@@ -496,5 +496,114 @@ bool can_prove(Expr e, const Scope<Interval> &bounds) {
     return is_const_one(e);
 }
 
+Simplify::ExprInfo::BitsKnown Simplify::ExprInfo::to_bits_known(const Type &type) const {
```

Comment:
A lot of this bitwise math is a little tricky to follow. Have you thrown this in an SMT solver? I think this should be verified